From c6427b5848178090146a463bbe723e2cf8b3391c Mon Sep 17 00:00:00 2001
From: classabbyamp <dev@placeviolette.net>
Date: Thu, 21 Aug 2025 11:52:27 -0400
Subject: [PATCH] backend/runit: fix race condition with dbus readiness

when logging in too fast on first boot, sometimes the dbus service isn't
ready when turnstile checks for the existence of the bus socket. This
seems to be enough to let that state settle before runit readiness is
indicated to turnstile.

fixes: #24
---
 backend/runit | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/backend/runit b/backend/runit
index 53d5757..15e69b8 100644
--- a/backend/runit
+++ b/backend/runit
@@ -82,6 +82,11 @@ cat << EOF > "${services_dir}/${ready_sv}/run"
 #!/bin/sh
 [ -r ./conf ] && . ./conf
 [ -n "\$core_services" ] && SVDIR=".." sv start \$core_services
+if [ -n "\$core_services" ]; then
+    until SVDIR=".." sv check \$core_services; do
+        :
+    done
+fi
 [ -p "$RUNIT_READY_PIPE" ] && printf "${services_dir}/${ready_sv}" > "$RUNIT_READY_PIPE"
 exec pause
 EOF
